src/Tree/RepositoryUtilsInterface.php

```diff
@@ -63,7 +63,7 @@ public function buildTree(array $nodes, array $options = []);
      *
      * @param object[] $nodes The nodes to build the tree from
      *
-     * @return array
+     * @return array<int, array<string, mixed>>
```

Comment:
cc @phansys the actual implementation returns objects

Reply:
Thank you for the feedback @ro0NL.
Could you please elaborate? I'm inferring this type based on the results of this test: https://github.com/doctrine-extensions/DoctrineExtensions/blob/157363b9fbe7e10aab5b2cb67765791bb59f43c0/tests/Gedmo/Tree/MaterializedPathORMRepositoryTest.php#L238-L295

The return value for this method on that test is something like this:
```
array(3) {
  [0]=>
  array(6) {
    ["id"]=>
    int(7)
    ["path"]=>
    string(9) "Drinks-7,"
    ["title"]=>
    string(6) "Drinks"
    ["level"]=>
    int(1)
    ["treeRootValue"]=>
    string(8) "Drinks-7"
    ["__children"]=>
    array(1) {
      [0]=>
      array(6) {
        ["id"]=>
        int(8)
        ["path"]=>
        string(18) "Drinks-7,Whisky-8,"
        ["title"]=>
        string(6) "Whisky"
        ["level"]=>
        int(2)
        ["treeRootValue"]=>
        string(8) "Drinks-7"
        ["__children"]=>
        array(1) {
          [0]=>
          array(6) {
            ["id"]=>
            int(9)
            ["path"]=>
            string(32) "Drinks-7,Whisky-8,Best Whisky-9,"
            ["title"]=>
            string(11) "Best Whisky"
            ["level"]=>
            int(3)
            ["treeRootValue"]=>
            string(8) "Drinks-7"
            ["__children"]=>
            array(0) {
            }
          }
        }
      }
    }
  }
  [1]=>
  array(6) {
    ["id"]=>
    int(1)
    ["path"]=>
    string(7) "Food-1,"
    ["title"]=>
    string(4) "Food"
    ["level"]=>
    int(1)
    ["treeRootValue"]=>
    string(6) "Food-1"
    ["__children"]=>
    array(2) {
      [0]=>
      array(6) {
        ["id"]=>
        int(3)
        ["path"]=>
        string(16) "Food-1,Fruits-3,"
        ["title"]=>
        string(6) "Fruits"
        ["level"]=>
        int(2)
        ["treeRootValue"]=>
        string(6) "Food-1"
        ["__children"]=>
        array(0) {
        }
      }
      [1]=>
      array(6) {
        ["id"]=>
        int(4)
        ["path"]=>
        string(20) "Food-1,Vegitables-4,"
        ["title"]=>
        string(10) "Vegitables"
        ["level"]=>
        int(2)
        ["treeRootValue"]=>
        string(6) "Food-1"
        ["__children"]=>
        array(2) {
          [0]=>
          array(6) {
            ["id"]=>
            int(5)
            ["path"]=>
            string(30) "Food-1,Vegitables-4,Carrots-5,"
            ["title"]=>
            string(7) "Carrots"
            ["level"]=>
            int(3)
            ["treeRootValue"]=>
            string(6) "Food-1"
            ["__children"]=>
            array(0) {
            }
          }
          [1]=>
          array(6) {
            ["id"]=>
            int(6)
            ["path"]=>
            string(31) "Food-1,Vegitables-4,Potatoes-6,"
            ["title"]=>
            string(8) "Potatoes"
            ["level"]=>
            int(3)
            ["treeRootValue"]=>
            string(6) "Food-1"
            ["__children"]=>
            array(0) {
            }
          }
        }
      }
    }
  }
  [2]=>
  &array(6) {
    ["id"]=>
    int(2)
    ["path"]=>
    string(9) "Sports-2,"
    ["title"]=>
    string(6) "Sports"
    ["level"]=>
    int(1)
    ["treeRootValue"]=>
    string(8) "Sports-2"
    ["__children"]=>
    array(0) {
    }
  }
}
```

Reply:
@phansys 

the contract takes object nodes: https://github.com/doctrine-extensions/DoctrineExtensions/blob/61c77d1fda1495aae144c06a4c6316d942110d16/src/Tree/RepositoryUtilsInterface.php#L64

the concrete implementaton appends object nodes: https://github.com/doctrine-extensions/DoctrineExtensions/blob/61c77d1fda1495aae144c06a4c6316d942110d16/src/Tree/RepositoryUtils.php#L167